### PR TITLE
Enabled warm-start flag for MOSEK10 and moved var initialization to _add_vars

### PR DESCRIPTION
**Pull request by Utkarsh-Detha**
## Summary/Motivation:
Issue #2613 was caused by a rare assert failure within MOSEK 10. That has since been fixed (please upgrade to **MOSEK 10.0.30**). 

But, while looking at the warmstart function in mosek_direct.py, I noticed that a parameter required to turn on initial solution utilization (only for v10 onwards) was missing. 

## Changes proposed in this PR:
- Enabed `mosek.iparam.mio_construct_sol` in warm_start function in mosek_direct.py
- Moved the code that passes variable initialization values from the `_warm_start` function to the `_add_vars` function.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by jsiirola — APPROVED** [review]
Looks good!

File: pyomo/solvers/plugins/solvers/mosek_direct.py [hunk 1]
```diff
@@ -361,6 +362,12 @@ def _add_vars(self, var_seq):
         self._pyomo_var_to_solver_var_map.update(zip(var_seq, var_ids))
         self._solver_var_to_pyomo_var_map.update(zip(var_ids, var_seq))
         self._referenced_variables.update(zip(var_seq, [0]*len(var_seq)))
+        # Initialize variables
+        for m_v, p_v in zip(var_ids, var_seq):
+            if p_v.value is not None:
+                for sol_type in mosek.soltype.values:
+                    self._solver_model.putxxslice(
+                        sol_type, m_v, m_v + 1, [(p_v.value)])
```
Comment (jsiirola) [inline pyomo/solvers/plugins/solvers/mosek_direct.py:370]: Question: by moving this to `_add_vars()`, does this prevent the variable values from being updated between subsequent solves in the case of the persistent solver?

Comment (Utkarsh-Detha) [reply pyomo/solvers/plugins/solvers/mosek_direct.py:370]: The same `task` object instance will be used over subsequent solves (when using mosek_persistent). So, the variable values from the previous solve will already be in that particular task. Pyomo would not have to pass them again. We only need to tell mosek if it should use those values or ignore them.

Comment (blnicho) [reply pyomo/solvers/plugins/solvers/mosek_direct.py:370]: @Utkarsh-Detha, reminder that this PR is waiting on a response from you to the above comment from @jsiirola.

Comment (Utkarsh-Detha) [reply pyomo/solvers/plugins/solvers/mosek_direct.py:370]: @jsiirola then I suppose we just leave it as is. I will undo the changes concerning this.

(Sorry about the slow response, I have been on holiday)

**Review by jsiirola — APPROVED** [review]
This looks good!

**Review by mrmundt — APPROVED** [review]

**Pull request merged**